### DISCUSSION TITLE
Why does a WebRTC video player sometimes fail to resume playback after being paused? How to fix?

### DISCUSSION THREAD

**Discussion by YiguiDing** [bluenviron/mediamtx#3706] [Questions]
### Question

```js
video.srcObject = mediaStream;
video.pause()
video.play() // why sometimes fail to resume playback
```

**reproduce**
```bash
mediamtx.exe
publish at  [http://localhost:8889/camera/publish]
pause and resume  video at  [http://localhost:8889/camera/]
```

[stream-player-api-javascript-play-and-pause-methods-not-working](https://community.cloudflare.com/t/stream-player-api-javascript-play-and-pause-methods-not-working/432195)

[Autoplay and Play function and play button not working when playing MediaStream](https://github.com/videojs/video.js/issues/7304)

**Comment by aler9** [comment]
Hello, unfortunately WebRTC is a complex system that requires to manually manage multiple components. Callling `pause` and `play` on the video tag is not enough since the underlying peer connection might be closed and in need of being recreated. As the README suggests, you can start by copying and pasting the web page embedded inside MediaMTX, that provides a ready-to-use class that implements all the logic:

https://github.com/bluenviron/mediamtx/blob/main/internal/servers/webrtc/reader.js

In order to play and pause the video, you can respectively instantiate and delete the class.
